### ISSUE TITLE
Build fails when task's dependencies array is empty

### ISSUE THREAD

**Issue by akwuh** [gulpjs/gulp#1454]
Hi,
This script fails due to an empty array of task dependencies.
```javascript
gulp.task('sass', [''], function() {
    return gulp.src('./scss/main.scss')
        .pipe(sass().on('error', sass.logError))
        .pipe(gulp.dest('./dist/css/'));
});
```

**Comment by TrySound** [comment]
@akwuh `''`dependency is not empty it's undefined. Use `null` or `undefined` values.

**Comment by TrySound** [comment]
@akwuh [Example](https://github.com/TrySound/case/blob/master/gulpfile.js#L64)

**Comment by contra** [comment]
This is fixed in 4.0 - there won't be any new 3.x releases, this task system has been completely redone to be much better. You can try it out with `npm install gulpjs/gulp#4.0`
